raritan/context.py

```diff
@@ -85,18 +86,37 @@ def set_data_reference(self, name: str, data_source) -> None:
         """
         self.data_references[name] = data_source
 
-    def get_data_reference(self, name: str):
+    def get_data_reference(self, name: str | list):
         """
         Gets a datasource reference or throws if one is not found.
 
         Parameters
         ----------
-        name: str
-            The name by which to access the data reference.
-        """
-        if name not in self.data_references.keys():
-            raise RuntimeError(f'Requested unloaded datasource, {name}. Was it included in @import_data?')
-        return self.data_references[name]
+        name: str | list
+            A name or list of names (or regexes) of data references to get.
+        """
+        if type(name) is str:
+            if name in self.data_references.keys():
+                return self.data_references[name]
+            pattern = re.compile(name)
+            matches = {}
+            for dataset_name, dataset in self.data_references.items():
+                if pattern.match(dataset_name):
+                    matches[dataset_name] = dataset
+            message = f'No data sources were named or matched, {name}. Was it included in @import_data?'
+            assert len(matches) > 0, message
+            return matches
+        if type(name) is list:
```

Comment:
**Non-Blocking**

Getting extra comments here telling us what's doing what would be nice.

Reply:
I added some more comments to a couple of the denser areas that were lacking. Thanks for the call out!